app/src/main/java/io/gnosis/safe/utils/StringUtils.kt

```diff
@@ -0,0 +1,56 @@
+package io.gnosis.safe.utils
+
+import android.content.Context
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.SpannableStringBuilder
+import android.text.style.ForegroundColorSpan
+import android.text.style.UnderlineSpan
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.utils.getColorCompat
+import pm.gnosis.svalinn.utils.ethereum.ERC67Parser
+import pm.gnosis.utils.*
+import java.math.BigInteger
+import io.gnosis.safe.R
+
+fun String.asMiddleEllipsized(boundariesLength: Int): String {
+    return if (this.length > boundariesLength * 2)
+        "${this.subSequence(0, boundariesLength)}...${this.subSequence(this.length - boundariesLength, this.length)}"
+    else this
+}
+
+fun String.underline(): CharSequence =
+    SpannableString(this).also {
+        it.setSpan(UnderlineSpan(), 0, length, 0)
+    }
+
+fun String.parseToBigIntegerOrNull(): BigInteger? =
+    nullOnThrow { parseToBigInteger() }
+
```

Comment:
Should we use the svalinn libs?

Reply:
I will check which module has this utility method and replace with the svalinn one.

Reply:
I think it was/is added in @biafra23 svalinn PR ... we should merge that tomorrow and maybe do a release